icevision/metrics/confusion_matrix/confusion_matrix.py

```diff
@@ -0,0 +1,153 @@
+            self.predicted_labels.extend(predicted_labels)
+
+    def finalize(self):
+        """Convert preds to numpy arrays and calculate the CM"""
+        assert len(self.target_labels) == len(self.predicted_labels)
+        label_ids = list(self.class_map._class2id.values())
+        self.confusion_matrix = sklearn.metrics.confusion_matrix(
+            y_true=self.target_labels,
+            y_pred=self.predicted_labels,
+            labels=label_ids,
+        )
+        if self.print_summary:
+            print(self.confusion_matrix)
+        self._reset()
+        return {"dummy_value_for_fastai": -1}
+
+    def plot(
+        self,
+        normalize: Optional[str] = None,
+        xticks_rotation="vertical",
+        values_format: str = None,
+        cmap: str = "PuBu",
+        figsize: int = 11,
+        **display_args,
+    ):
+        """
+        A handle to plot the matrix in a jupyter notebook, potentially this could also be passed to save_fig
+        """
+        if normalize not in ["true", "pred", "all", None]:
+            raise ValueError(
+                "normalize must be one of {'true', 'pred', " "'all', None}"
+            )
+        # properly display ints and floats
+        if values_format is not None:
+            values_format = ".2f" if normalize else "d"
+
+        cm = self._maybe_normalize(self.confusion_matrix, normalize)
+        labels_named = self.class_map._id2class
+        cm_display = sklearn.metrics.ConfusionMatrixDisplay(
+            cm, display_labels=labels_named
+        )
+        figure = cm_display.plot(
+            xticks_rotation=xticks_rotation,
+            cmap=cmap,
+            values_format=values_format,
+            **display_args,
+        ).figure_
+        figure.set_size_inches(figsize, figsize)
+        plt.close()
+        return figure
+
+    def _fig2img(self, fig):
+        # TODO: start using fi2img from icevision utils
+        """Converts matplotlib figure object to PIL Image for easier logging. Writing to buffer is necessary
+        to avoid wandb cutting our labels off. Wandb autoconvert doesn't pass the `bbox_inches` parameter so we need
+        to do this manually."""
+        buf = io.BytesIO()
+        fig.savefig(buf, bbox_inches="tight")
+        buf.seek(0)
+        return PIL.Image.open(buf)
+
+    def _maybe_normalize(self, cm, normalize):
+        """This method is copied from sklearn. Only used in plot_confusion_matrix but we want to be able
+        to normalize upon plotting."""
+        with np.errstate(all="ignore"):
+            if normalize == "true":
+                cm = cm / cm.sum(axis=1, keepdims=True)
+            elif normalize == "pred":
+                cm = cm / cm.sum(axis=0, keepdims=True)
+            elif normalize == "all":
+                cm = cm / cm.sum()
+        cm = np.nan_to_num(cm)
+        return cm
+
+    def log(self, logger_object) -> None:
+        if isinstance(logger_object, pl_loggers.WandbLogger):
```

Comment:
these soft imports get complicated here right? I'm also not sure rn what is the best way to handle this, but this function will error out if lightning is not installed

Reply:
here we can do the following:
```python
if SoftDependencies.pytorch_lightining:
    if isinstance(logger_object, pl.loggers.WandbLogger):
        # do the stuff
```

Doesn't look good and I don't think it's a sustainable solution, but works for now?

Reply:
i disabled logging at all cause it is not supported in current metric design anyway